yt/geometry/oct_container.pyx

```diff
@@ -728,17 +728,196 @@ cdef class OctreeContainer:
+        domain_id : int, optional
+            The domain to perform the selection over
+
+        Returns
+        -------
+        oct_inds : int64 ndarray (nocts*8, )
+            The on-domain index of the octs containing each cell
+        cell_inds : uint8 array (nocts*8, )
+            The index of the cell in its parent oct
+
+        Note
+        ----
+        oct_inds/cell_inds
+        """
+        if num_octs == -1:
+            num_octs = selector.count_octs(self, domain_id)
+
+        cdef NeighbourCellIndexVisitor visitor
+
+        cdef np.uint8_t[:] cell_inds
+        cdef np.int64_t[:] oct_inds
+
+        cell_inds = np.full(num_octs*4**3, 8, dtype=np.uint8)
+        oct_inds = np.full(num_octs*4**3, -1, dtype=np.int64)
+
+        visitor = NeighbourCellIndexVisitor(self, -1)
+        visitor.cell_inds = cell_inds
+        visitor.domain_inds = oct_inds
+
+        self.visit_all_octs(selector, visitor)
+
+        return np.asarray(oct_inds), np.asarray(cell_inds)
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    @cython.cdivision(True)
+    def fill_level_with_domain(
+                   self, int level,
+                   np.uint8_t[:] levels,
+                   np.uint8_t[:] cell_inds,
+                   np.int64_t[:] file_inds,
+                   np.int32_t[:] domains,
+                   dict dest_fields, dict source_fields,
+                   np.int32_t domain,
+                   np.int64_t offset = 0
+                   ):
+        """Similar to fill_level but accepts a domain argument.
+
+        This is particularly useful for frontends that have buffer zones at CPU boundaries.
+        These buffer oct cells have a different domain than the local one and
+        are usually not read, but one has to read them e.g. to compute ghost zones.
+        """
+        cdef np.ndarray[np.float64_t, ndim=2] source
+        cdef np.ndarray[np.float64_t, ndim=1] dest
+        cdef int i, count
+
+        for key in dest_fields:
+            dest = dest_fields[key]
+            source = source_fields[key]
+            count = 0
+            for i in range(levels.shape[0]):
+                if levels[i] != level or domains[i] != domain: continue
+                count += 1
+                if file_inds[i] < 0:
+                    dest[i + offset] = np.nan
+                else:
+                    dest[i + offset] = source[file_inds[i], cell_inds[i]]
+        return count
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    @cython.cdivision(True)
+    def file_index_octs_with_ghost_zones(
+            self, SelectorObject selector, int domain_id,
+            int num_cells = -1):
+        """Similar as file_index_octs, but return as well the level, cell index,
+        file index and domain of the neighbouring cells.
+
+        Arguments
+        ---------
+        selector : SelectorObject
+            The selector object. It is expected to select all cells for a selected oct.
```

Comment:
Would this make sense ? I'm trying to avoid the repetition of "select"
```suggestion
            The selector object. It is expected to select all cells for a given oct.
```

Reply:
Done.